core/templates/dev/head/player/PlayerServices.js

```diff
@@ -357,10 +378,30 @@ oppia.factory('oppiaPlayerService', [
     getGadgetPanelsContents: function() {
       return angular.copy(_exploration.skin_customizations.panels_contents);
     },
+    getInteractionThumbnailSrc: function(stateName) {
+      // TODO(sll): unify this with the 'choose interaction' modal in
+      // state_editor_interaction.html.
+      var interactionId = _exploration.states[stateName].interaction.id;
+      if (!interactionId) {
+        return '';
+      } else {
+        return (
+          '/extensions/interactions/' + interactionId + '/static/' +
+          interactionId + '.png');
+      }
+    },
+    getInteractionInstructions: function(stateName) {
+      var interactionId = _exploration.states[stateName].interaction.id;
+      if (!interactionId) {
+        return '';
+      } else {
+        return INTERACTION_SPECS[interactionId].instructions;
+      }
+    },
     isInteractionInline: function(stateName) {
       var interactionId = _exploration.states[stateName].interaction.id;
       return (
-        interactionId &&
+        !interactionId ||
```

Comment:
In the editor preview mode, a null interaction (which happens when one isn't specified for a state) takes the form of a single line of red text ("No interaction specified"). It seems a bit gratuitous to put this in its own card.

Do you think it needs a comment? Future developers will only see this code, and not the fact that it has changed.